src/Core/src/VisualDiagnostics/VisualDiagnostics.cs

```diff
@@ -13,16 +13,19 @@ namespace Microsoft.Maui
 	public static class VisualDiagnostics
 	{
 		static ConditionalWeakTable<object, SourceInfo> sourceInfos = new ConditionalWeakTable<object, SourceInfo>();
+		static Lazy<bool> isVisualDiagnosticsEnvVarSet  = new Lazy<bool>(() => Environment.GetEnvironmentVariable("ENABLE_XAML_DIAGNOSTICS_SOURCE_INFO") is { } value && value == "1");
+
+		static public bool IsEnabled => DebuggerHelper.DebuggerIsAttached || isVisualDiagnosticsEnvVarSet.Value;
```

Comment:
Does this need to be a public variable? Since it's not settable and not used outside of this repo I'm not sure if it's needed to be accessible.

Reply:
Good point!

Reply:
Actually, this is defined in Core, but Control.Core needs to access it.